File: src/backend/Intelibill.Domain/Entities/Customer.cs
using Intelibill.Domain.Common;

namespace Intelibill.Domain.Entities;

public sealed class Customer : BaseEntity
{
    public Guid ShopId { get; private set; }
    public string Name { get; private set; } = string.Empty;
    public string PhoneNumber { get; private set; } = string.Empty;
    public string? Address { get; private set; }
    public bool IsActive { get; private set; }
    public decimal CreditLimit { get; private set; }

    private Customer() { }

    public static Customer Create(
        Guid shopId,
        string name,
        string phoneNumber,
        string? address,
        bool isActive = true,
        decimal creditLimit = 0m)
    {
        return new Customer
        {
            ShopId = shopId,
            Name = name.Trim(),
            PhoneNumber = phoneNumber.Trim(),
            Address = NormalizeOptional(address),
            IsActive = isActive,
            CreditLimit = creditLimit,
        };
    }

    public void Update(
        string name,
        string phoneNumber,
        string? address,
        bool isActive,
        decimal creditLimit)
    {
        Name = name.Trim();
        PhoneNumber = phoneNumber.Trim();
        Address = NormalizeOptional(address);
        IsActive = isActive;
        CreditLimit = creditLimit;
    }

    private static string? NormalizeOptional(string? value)
    {
        if (string.IsNullOrWhiteSpace(value))
            return null;

        return value.Trim();
    }
}

File: src/mobile/android/intelibill_mobile/lib/src/features/customers/domain/entities/customer.dart
import 'package:equatable/equatable.dart';

class Customer extends Equatable {
  const Customer({
    required this.customerId,
    required this.name,
    required this.phoneNumber,
    this.address,
    required this.isActive,
    this.outstandingDue = 0.0,
  });

  final String customerId;
  final String name;
  final String phoneNumber;
  final String? address;
  final bool isActive;
  final double outstandingDue;

  @override
  List<Object?> get props => [
    customerId,
    name,
    phoneNumber,
    address,
    isActive,
    outstandingDue,
  ];
}
